### DISCUSSION TITLE
How to disable Wifi after boot?

### DISCUSSION THREAD

**Discussion by vizete** [MiczFlor/RPi-Jukebox-RFID#2087] [Q&A] [answered]
Hi,

I set up my very first Jukebox. Most of the things working and must say that is really a great project!

Currently, version 2.4 is running on my Raspberry Pi 3B+. (see details at the end of this post)

I configured a button to dis-/enable the Wifi by using the function 'functionCallToggleWifi' as described in the 'GPIO CONTROL' section. The box is configured to connect to a Wifi named 'PhonieHost'. I'm starting a hotspot with the same name on my notebook on demand. So I can connect to the box no matter if there is any known network around me.

**How can I configure the box in a way that the Wifi is disabled after every boot process?** Currently, the Wifi is enabled after booting. (I'm not sure if it returns to the state when shutting down or it is enabled as default value.)
I tried to add 'dtoverlay=disable-wifi' to the /boot/config.txt but then I can't enable the adapter via the configured button.
The connection is only used when configuring the Jukebox or adding some new content. So I prefer the box to be disconnected as default state and connect it on demand only.

Thanks and best regards.

=== System ===
OS Distribution: Raspbian
Description: Raspbian GNU/Linux 10 (buster)
Release: 10
Codename: buster
=== Phoniebox Setup ===
Version: 2.4 - abcf6e92 - master
Edition: Classic edition (barebones)

**Comment by AlvinSchiller** [comment]
Hi,
i've tried your case on Buster and Bullseye, main and develop and can’t reproduce it. After triggering the "toggleWifi" Command the wifi is de-/activated and after Soft/Cold reset the state is still preserved.
I tried it with a RFID Card though and not a button, but this will trigger the same logic under the hood, so it should not behave different.

Do you maybe have more information about your installation? Maybe post your "./settings/PhonieboxInstall.conf" (mask passwords, etc.)?

**Reply by AlvinSchiller** [reply]
As additional Information: 
The Phoniebox already has a feature which will create a hotspot automatically if no known network is available.
Maybe that's relevant for you, as it would enable you to connect with any device to your box :)
This feature is currently been merged into the main installer script (https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2086)

**Reply by vizete** [reply]
Hello Alvin,

unfortunately, I must have expressed myself in a misleading way. What I'm looking for is that the Wifi is disabled after booting. No matter what the state was before shutting the box down. (the state should **not** be preserved)

The Background: I want to prevent the Wifi from being accidentally active. E.g. because I forgot to shut it down in the last session.

Thanks

**Reply by AlvinSchiller** [reply, marked as answer]
Ah ok, i missunderstood that part.

You can try and add the command `rfkill block wifi || true` to the file `/etc/rc.local` file, right before the exit command.
That way the wifi should be disabled on boot and can be activated later again (the same command is used internally in phoniebox).

It worked in my tests.

**Reply by AlvinSchiller** [reply]
Did you find a solution?

**Reply by vizete** [reply]
Sorry for the (very) late reply. Yes, your suggestion worked for me. 

- I added the function 'functionCallToggleWifi' in '~/RPi-Jukebox-RFID/settings/gpio_settings.ini'
- I checked/disabled 'dtoverlay=disable-wifi' in '/boot/config.txt'
- and finally added 'rfkill block wifi || true' to '/etc/rc.local' (as you suggested)

Now the Wifi is disabled after each reboot and I can enable it on demand by pressing the defined button 'gpio_settings.ini'.

Thanks a lot,
Daniel
